components/tidb_query/src/codec/mysql/json/json_keys.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::super::Result;
+use super::path_expr::PathExpression;
+use super::Json;
+
+impl Json {
+    pub fn keys(&self, path_expr_list: &[PathExpression]) -> Result<Option<Json>> {
+        if !path_expr_list.is_empty() {
+            if path_expr_list.len() > 1 {
+                return Err(box_err!("Incorrect parameter count"));
```

Comment:
Better to contain the `path_expr_list ` to error message to help locate problem.

Reply:
This error is an `Incorrect parameter count`, may be we just need to show the length of 
 `path_expr_list`?

Reply:
> Would you mind implement `json_keys` in `builtin_json.rs` or `impl_json.rs`?

Yes,I will

Reply:
done